common/include/pcl/common/impl/random.hpp

```diff
@@ -40,14 +40,10 @@
 #ifndef PCL_COMMON_RANDOM_HPP_
 #define PCL_COMMON_RANDOM_HPP_
 
-#include <boost/version.hpp>
-#include <pcl/pcl_macros.h>
-
 /////////////////////////////////////////////////////////////////////////////////////////////////////////
 template <typename T>
 pcl::common::UniformGenerator<T>::UniformGenerator(T min, T max, pcl::uint32_t seed)
   : distribution_ (min, max)
-  , generator_ (rng_, distribution_)
 {
   parameters_ = Parameters (min, max, seed);
   if(parameters_.seed != -1)
```

Comment:
Just saw a compiler warning related to this:
```
common/include/pcl/common/impl/random.hpp:49:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if(parameters_.seed != -1)
```
Even I didn't touched here anything: Should we fix this warning in this PR (and how?) or in a separate?

Reply:
We can fix it here. Just `static_cast<pcl::uint32_t>(-1)`.

Reply:
Just wondering about this:
https://github.com/PointCloudLibrary/pcl/blob/b4c785bf32b392b4570201e8fe9f97395bd98ba1/common/include/pcl/common/random.h#L163-L186

One constructor of `NormalGenerator` has as default seed `1` during the other one (via `Parameters`) has `-1` as default seed.

Reply:
I think `-1` should be the default everywhere. However, thinking more about this, wouldn't it be better if `-1` meant "seed with random device" instead of "not seed at all".

Reply:
Moved discussion to #2985. I think it is better, if this behavior changes are not part of this PR.